slack/views.py

```diff
@@ -26,24 +29,25 @@ def slash_command(request):
     @param request the request from slack containing the slash command
     @return: return a HTTP response to indicate the request was handled
     """
-    report = request.POST.get('text')
-
-    if not report:
-        return HttpResponse("It looks like you forgot to provide any details 🤔 \
-        \n\nCan you try again with something like `/incident something has happened`?")
 
     user_id = request.POST.get('user_id')
+    trigger_id = request.POST.get('trigger_id')
+    report = request.POST.get('text')
 
-    Incident.objects.create_incident(
-        report=report,
-        reporter=user_id,
-        report_time=datetime.now(),
+    dialog = Dialog(
+        title="Report an Incident",
+        submit_label="Report",
+        elements=[
+            Text(label="Report", name="report", placeholder="What's the tl;dr?", value=report),
+            TextArea(label="Summary", name="summary", optional=True, placeholder="Can you share any more details?"),
+            TextArea(label="Impact", name="impact", optional=True, placeholder="Who or what might be affected?", hint="Think about affected people, systems, and processes"),
+            SelectFromUsers(label="Lead", name="lead", optional=True),
+            SelectWithOptions([(i, s.capitalize()) for i, s in Incident.SEVERITIES], label="Severity", name="severity", optional=True)
+        ]
     )
 
-    incidents_channel_ref = channel_reference(settings.INCIDENT_CHANNEL_ID)
-    text = f"Thanks for raising the incident 🙏\n\nHead over to {incidents_channel_ref} to complete the report and/or help deal with the issue"
```

Comment:
I think it would be nice to keep this message, so the reporter has a link to click to the incidents channel, as well as a message to say thank you for raising the incident?

Reply:
That's a good shout, although I think it'll need to move to the section of code that handles dialog submissions.   If it remains here, users will get that message even if they cancel the report.